Add possibility to configure timeout for remote process

t-h-e opened the pull request:
[NOTE]: # ( Describe your changes in detail, why is this change required? )
[NOTE]: # ( Explain large or complex code modifications. )
[NOTE]: # ( If it fixes an open issue, please add "Fixes #XXX" )

Addresses issue https://github.com/keepassxreboot/keepassxc/issues/11234

Does NOT break existing configurations, as default values are used.

## Screenshots
[TIP]:  # ( Do not include screenshots of your actual database! )

![image](https://github.com/user-attachments/assets/7020b7c0-e351-4eff-bc82-ba79c216e841)

## Testing strategy
[NOTE]: # ( Please describe in detail how you tested your changes. )
[TIP]:  # ( We expect new code to be covered by unit tests and documented with doc blocks! )


## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ New feature (change that adds functionality)

t-h-e committed:
Add possibility to configure timeout for remote process

droidmonkey commented on src/gui/remote/RemoteHandler.cpp:
```diff
@@ -67,7 +67,7 @@ RemoteHandler::RemoteResult RemoteHandler::download(const RemoteParams* params)
         }
 
-        bool finished = remoteProcess->waitForFinished(10000);
+        bool finished = remoteProcess->waitForFinished(params->downloadTimeout * 1000);
```
Instead of doing math here I recommend doing that at the setting of the downloadTimeout (which could be called `downloadTimeoutMsec` to be explicit). In general, I am trying to keep all "display math" at the GUI layer and the backend has well defined parameters.

droidmonkey commented on src/gui/remote/RemoteHandler.cpp:
```diff
@@ -118,7 +118,7 @@ RemoteHandler::RemoteResult RemoteHandler::upload(const QString& filePath, const
         }
 
-        bool finished = remoteProcess->waitForFinished(10000);
+        bool finished = remoteProcess->waitForFinished(params->uploadTimeout * 1000);
```
same here

droidmonkey commented on src/gui/remote/DatabaseSettingsWidgetRemote.cpp:
```diff
@@ -46,8 +46,10 @@ DatabaseSettingsWidgetRemote::DatabaseSettingsWidgetRemote(QWidget* parent)
     connect(m_ui->nameLineEdit, &QLineEdit::textChanged, setModified);
     connect(m_ui->downloadCommand, &QLineEdit::textChanged, setModified);
     connect(m_ui->inputForDownload, &QPlainTextEdit::textChanged, setModified);
+    connect(m_ui->downloadTimeout, QOverload<int>::of(&QSpinBox::valueChanged), setModified);
```
Since you don't use the value from this signal you could just use `textChanged` to avoid the awkward QOverload

t-h-e replied:
I'm not sure why but the Ubuntu build does not agree
```
/opt/buildagent/work/c401303cba1b4098/src/gui/remote/DatabaseSettingsWidgetRemote.cpp: In constructor ‘DatabaseSettingsWidgetRemote::DatabaseSettingsWidgetRemote(QWidget*)’:
/opt/buildagent/work/c401303cba1b4098/src/gui/remote/DatabaseSettingsWidgetRemote.cpp:49:50: error: ‘textChanged’ is not a member of ‘QSpinBox’
   49 |     connect(m_ui->downloadTimeoutSec, &QSpinBox::textChanged, setModified);
      |                                                  ^~~~~~~~~~~
/opt/buildagent/work/c401303cba1b4098/src/gui/remote/DatabaseSettingsWidgetRemote.cpp:52:48: error: ‘textChanged’ is not a member of ‘QSpinBox’
   52 |     connect(m_ui->uploadTimeoutSec, &QSpinBox::textChanged, setModified);
      |                                                ^~~~~~~~~~~
make[2]: *** [src/CMakeFiles/keepassxc_gui.dir/build.make:1254: src/CMakeFiles/keepassxc_gui.dir/gui/remote/DatabaseSettingsWidgetRemote.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:677: src/CMakeFiles/keepassxc_gui.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
Process exited with code 2
```

droidmonkey replied:
Ugh it was first introduced in qt 5.14. Sorry for the bad recommendation, have to revert back to valueChanged

t-h-e committed:
Address review comments

- move calculations to GUI layer
- add unit to downloadTimeout and uploadTimeout
